internal/model/command.go

```diff
@@ -193,6 +210,130 @@ func (c ExecutableCommand[F]) GetFlagValues(cmd *cobra.Command) (*F, error) {
+func runWithVersionFromWorkflowFile(cmd *cobra.Command, args []string) error {
+	ctx := cmd.Context()
+	logger := log.From(ctx)
+
+	wf, wfPath, err := utils.GetWorkflow()
+	if err != nil {
+		return fmt.Errorf("failed to load workflow file: %w", err)
+	}
+
+	// If the workflow file doesn't exist, or we're running locally, simply run the command normally with the existing version of the CLI
+	if wf == nil { // TODO: uncomment when done testing locally || env.IsLocalDev() {
+		return nil
+	}
+
+	currentVersion := events.GetSpeakeasyVersionFromContext(ctx)
+
+	// Try to migrate existing workflows
+	if wf.SpeakeasyVersion == "" {
+		if ghPinned := env.PinnedVersion(); ghPinned != "" {
+			wf.SpeakeasyVersion = workflow.Version(ghPinned)
+			wf.VersionLocked = true
+		} else {
+			wf.SpeakeasyVersion = workflow.Version(currentVersion)
+		}
+
+		_ = updateWorkflowFile(wf, wfPath)
+	}
+
+	if wf.SpeakeasyVersion != "" && wf.SpeakeasyVersion != "latest" {
+		artifactArch := ctx.Value(updates.ArtifactArchContextKey).(string)
+		desiredVersion := wf.SpeakeasyVersion.String()
+
+		// If the desired version is the same as the currently running version of the CLI, just run the command
+		// Caveat: this means we might not auto-upgrade in certain cases
+		if desiredVersion == currentVersion {
+			return nil
+		}
+
+		newerVersion, err := updates.GetNewerVersion(artifactArch, desiredVersion)
+		if err != nil {
+			return err
+		}
+
+		if wf.VersionLocked || newerVersion == nil {
+			if err := runWithVersion(cmd, artifactArch, desiredVersion); err != nil {
+				return err
+			}
+		} else {
+			logger.PrintfStyled(styles.DimmedItalic, "Newer Speakeasy version found (%s => %s). Attempting auto-upgrade\n", desiredVersion, newerVersion.String())
+
+			err = runWithVersion(cmd, artifactArch, newerVersion.String())
+			if err != nil {
+				msg := fmt.Sprintf("Failed to auto-upgrade to version %s: %s\n", newerVersion.String(), err.Error())
+				if err = log.SendToLogProxy(ctx, log.LogProxyLevelError, msg, nil); err != nil {
+					logger.Errorf("Failed to send log to LogProxy: %v\n", err)
+				}
+
+				logger.PrintfStyled(styles.DimmedItalic, msg)
+				logger.PrintfStyled(styles.DimmedItalic, "Rerunning with version defined in workflow.yaml: %s\n", desiredVersion)
+
+				if err := runWithVersion(cmd, artifactArch, desiredVersion); err != nil {
+					return err
+				}
+			} else {
+				wf.SpeakeasyVersion = workflow.Version(newerVersion.String())
+				if err := updateWorkflowFile(wf, wfPath); err != nil {
+					return err
+				}
+				logger.Successf("\nAuto-upgrade successful! Updated workflow.yaml/speakeasyVersion to %s\n", newerVersion.String())
+			}
+		}
+
+		// Exit here to prevent the command from running twice
+		os.Exit(0)
+	}
+
+	return nil
+}
+
+func runWithVersion(cmd *cobra.Command, artifactArch, desiredVersion string) error {
+	ctx := cmd.Context()
+
+	vLocation, err := updates.InstallVersion(ctx, desiredVersion, artifactArch, 30)
```

Comment:
Just looking at the code I think `vLocation` is going to be missing the `.exe` suffix here on windows